twenty-first/src/tip5/mod.rs

```diff
@@ -1067,6 +1089,50 @@ pub(crate) mod tests {
         assert_eq!(Tip5::offset_fermat_cube_map(255), 255);
     }
 
+    /// Ensure that the claims made in [`Tip5::mds_generated`] are true.
+    ///
+    /// In particular, `BFieldElement::Add` internally uses the equality
+    /// `a + b = a - (p - b)`. If `a` is a degenerate representation (i.e., the
+    /// Montgomery representation is larger than or equal to
+    /// [`BFieldElement::P`]), then a small enough `b` makes the result of the
+    /// addition non-degenerate by removing the “surplus” from `a`. In
+    /// particular, this correction will happen if the following inequality
+    /// holds:
+    ///              p - b > u64::MAX - p
+    ///   ↔          p - b > u32::MAX
+    ///   ↔   p - u32::MAX > b
+    ///
+    /// While it’s not particularly beautiful to depend on such implementation
+    /// details, Ferdinand is too scared to change the implementation of Tip5.
+    /// If you change the behavior of `BFieldElement::Add`, please make sure
+    /// that [`Tip5`] is not breaking.
+    ///
+    /// The test [`round_constants_correct_degenerate_lhs_when_adding`] makes
+    /// sure that all round constants have the required property.
+    ///
+    /// See also: https://www.hyrumslaw.com/. Sorry about that.
+    #[proptest]
+    fn adding_degenerate_lhs_and_small_enough_rhs_makes_lhs_non_degenerate(
+        #[strategy(BFieldElement::P..)] degenerate_left: u64,
+        #[strategy(0..BFieldElement::P - u64::from(u32::MAX))] raw_right: u64,
```

Comment:
I don't see how this strategy is correct.

I would naïvely expect the variables that take the values of `a` and `b` in the above text to be called `a` and `b`. But now there is a renaming of variables and algebraic manipulations going on, and that's too much strain for my little brain.